src/idpi/operators/destagger.py

```diff
@@ -1,39 +1,68 @@
 """algorithm for destaggering a field."""
+# Standard library
+from typing import Literal
+
 # Third-party
+import numpy as np
 import xarray as xr
 
 
-def destagger(field: xr.DataArray, dim: str) -> xr.DataArray:
+def _dstgr(a: np.ndarray):
+    t = a.copy()
+    t[..., 1:] = 0.5 * (a[..., :-1] + a[..., 1:])
+    return t
+
+
+def _dstgr_z(a: np.ndarray):
+    return 0.5 * (a[..., :-1] + a[..., 1:])
```

Comment:
I am not sure about this comment, but here we are supposedly passing only 1D arrays, i.e. the only dimension that has not been broadcasted from the apply_ufunc ? that means, do we still need/want the `...` ?

Reply:
I had the same thought but I checked and the array is just transposed.